code/game/machinery/doors/firedoor.dm

```diff
@@ -135,10 +136,35 @@
 		return
 
 	if(density)
+		being_held_open = TRUE
 		open()
+		handle_held_open_adjacency(user)
 	else
 		close()
 
+/obj/machinery/door/firedoor/try_to_crowbar_secondary(obj/item/I, mob/user)
+	if(welded || operating)
+		return
+
+	if(density)
+		open()
+	else
+		close()
+
+/obj/machinery/door/firedoor/proc/handle_held_open_adjacency(mob/user)
+	var/mob/living/living_user = isliving(user) ? user : null
+	if(!Adjacent(user) || !isliving(user) || !living_user.is_standing())
+		being_held_open = FALSE
+		close()
+		return
+
+	if (!do_after(living_user, 30 SECONDS, src, progress=FALSE, extra_checks = CALLBACK(living_user, /mob/living/.proc/is_standing)))
+		handle_held_open_adjacency(living_user)
```

Comment:
I don't understand, what does this actually do and why is there a 30 second do_after regarding crowbars now?

Reply:
On second thought, this is kinda stupid. I'll implement a proper callback datum later.